src/openstack_cloud/openstack_manager.py

```diff
@@ -805,8 +805,17 @@ def _create_runner(args: _CreateRunnerArgs) -> None:
                         instance_config.name,
                     )
                     conn.delete_server(name_or_id=instance_config.name, wait=True)
+                    try:
+                        conn.delete_keypair(instance_config.name)
```

Comment:
Should it ignore errors if keypair does not exist?

Reply:
I think this would be thrown as an SDKException and we would log an exception. I think this should be sufficient (I basically reused the same code we have in other parts of this module for cleaning up keypairs).